src-backend/api/tests/test_generator.py

```diff
@@ -0,0 +1,274 @@
+        self.element.numeric = 'DIALPAD'
+        self.elementElement = self.generator.generate(ElementTree.Element('test'))
+
+        print self.elementElement.attrib
+
+        assert_true('numeric' in self.elementElement.attrib)
+        assert_equals(self.elementElement.attrib['numeric'], self.element.numeric)
+
+    def test_element_has_choices(self):
+        self.element.choices = 'left,right,center'
+        self.elementElement = self.generator.generate(ElementTree.Element('test'))
+
+        assert_true('choices' in self.elementElement.attrib)
+        assert_equals(self.elementElement.attrib['choices'], self.element.choices)
+
+
+class ProtocolBuilderTestCase(TestCase):
+    def setUp(self):
+        self.test_user = User.objects.create_user(
+            'TestUser',
+            'test@sanaprotocolbuilder.me',
+            'testpassword'
+        )
+        self.test_user.save()
+
+        self.procedure = Procedure.objects.create(
+            author='TestUser',
+            title='Burns',
+            owner=self.test_user
+        )
+
+        page1 = Page.objects.create(
+            display_index=0,
+            procedure=self.procedure
+        )
+        page2 = Page.objects.create(
+            display_index=1,
+            procedure=self.procedure
+        )
+        page3 = Page.objects.create(
+            display_index=2,
+            procedure=self.procedure
+        )
+        page4 = Page.objects.create(
+            display_index=3,
+            procedure=self.procedure
+        )
+
+        Element.objects.create(
+            display_index=0,
+            eid='burns',
+            element_type='MULTI_SELECT',
+            concept='BURNS',
+            question='Select One or More of the Following:',
+            answer='',
+            choices='On the head,On the chest,Loins,Inhalation burns,Chemical,In an enclosure,Patient Unconscious,Epileptic Patient, Diabetic Patient',  # noqa
+            page=page1
+        )
+
+        Element.objects.create(
+            display_index=1,
+            eid='duration',
+            element_type='ENTRY',
+            concept='COMPLAINT DURATION',
+            question='Enter Complaint Duration in Days:',
+            answer='',
+            numeric='DIALPAD',
+            page=page2
+        )
+
+        Element.objects.create(
+            display_index=2,
+            eid='COMMENTS',
+            element_type='ENTRY',
+            concept='BURNS',
+            question='Other comments:',
+            answer='',
+            page=page3
+        )
+
+        Element.objects.create(
+            display_index=3,
+            eid='picture',
+            element_type='PICTURE',
+            concept='COMPLAINT PICTURE',
+            question='Add a Picture:',
+            answer='',
+            page=page4
+        )
+
+        self.procedure.save()
+
+    def test_generates_tree(self):
+        tree = ProtocolBuilder._generateTree(self.test_user, self.procedure.id)
```

Comment:
I think this method should either be public or we only test the pre/post conditions of any public methods that will call this internally (http://c2.com/cgi/wiki?InappropriateIntimacy)